### DISCUSSION TITLE
Sudden Death fine tuning

### DISCUSSION THREAD

**Discussion by AN0NCER** [ppy/osu#19748] [Ideas]
I propose to add to the game a fine-tuning of Sudden Death, the ability to select the number of combos, after which it will stop working. That is, if the user chooses **100x** combos, then if he makes a mistake at **50x** combos, he will lose, and if at **102x**, then the game will continue no matter how many mistakes he makes further.

For Osu Lazzer!

**Comment by mk56-spn** [comment]
i can definitely see this being of use for people trying to get long attempts on maps from the start, semi niche but useful

(ps. everything on this github repo is for lazer so no need to mention it :P, stable being featured locked and all especially)

**Reply by AN0NCER** [reply]
As far as I remember, Sudden Death still doesn't have a difficulty multiplier. I think that it should not be difficult to implement

**Reply by mk56-spn** [reply]
Sudden death does not change the map or visibility in any shape or form  so I'm afraid that's unlikely to happen. What difficulty it does add is purely psychological in its nature

**Reply by AN0NCER** [reply]
Probably misunderstood me, I meant that due to the fact that this mod does not have a difficulty multiplier, it will be easier to make it. (Just my guesses)

**Reply by nagi-desuuu** [reply]
Nope, still not going to happen.
Sudden Death only allows you to pass if you have an unbroken combo, and should NOT be able to provide you with an additional multiplier.
Besides, it's been like this for a long time now. If we mess with the settings, the old replays are going to be doomed in terms of scores.

**Reply by AN0NCER** [reply]
**Stop!** I'm not asking for a score multiplier!!

**Comment by AN0NCER** [comment]
Here is the implementation of this mod from me. If something can be taken out as a separate mod, but it seems to me that it’s better this way ...

```cs
public abstract class ModFailCondition : Mod, IApplicableToHealthProcessor, IApplicableFailOverride, IApplicableToScoreProcessor
    {

        public override Type[] IncompatibleMods => new[] { typeof(ModNoFail), typeof(ModRelax), typeof(ModAutoplay) };

        [SettingSource("Restart on fail", "Automatically restarts when failed.")]
        public BindableBool Restart { get; } = new BindableBool();

        [SettingSource("Restart if < combo", "Disable mod after combo.")]
        public BindableBool EnableScore { get; } = new BindableBool();

        [SettingSource("Combo", "", SettingControlType = typeof(SettingsNumberBox))]
        public Bindable<int?> Seed { get; } = new Bindable<int?>
        {
            Default = 0,
            Value = 0
        };

        public bool IsCombo = false;

        public virtual bool PerformFail() => true;

        public virtual bool RestartOnFail => Restart.Value;

        public virtual bool DisableOnScore => EnableScore.Value;

        public void ApplyToHealthProcessor(HealthProcessor healthProcessor)
        {
            if (DisableOnScore)
                healthProcessor.FailConditions += FF;
            else
                healthProcessor.FailConditions += FailCondition;
        }

        public bool FF(HealthProcessor healthProcessor, JudgementResult result)
        {

            if (Seed.Value <= Combo.Value)
                IsCombo = true;

            if (IsCombo)
                return false;
            else
                return FailCondition(healthProcessor, result);
        }

        protected abstract bool FailCondition(HealthProcessor healthProcessor, JudgementResult result);

        protected readonly BindableInt Combo = new BindableInt();

        public void ApplyToScoreProcessor(ScoreProcessor scoreProcessor) =>
            Combo.BindTo(scoreProcessor.Combo);

        public ScoreRank AdjustRank(ScoreRank rank, double accuracy) =>
            rank;
    }
```

How he looks now:

![Image](https://raw.githubusercontent.com/AN0NCER/AN0NCER/main/Discussion/osu_sun_deth.png)

**Reply by mk56-spn** [reply]
i wouldnt put math symbols in UX, rephrase it to something like " restart if combo is lower than: " and delete the word combo from under it
